core/src/main/java/io/grpc/internal/AbstractServerImplBuilder.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2020 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.internal;
+
+import io.grpc.ForwardingServerBuilder;
+import io.grpc.ServerBuilder;
+
+/**
+ * Temporary shim for {@link io.grpc.ForwardingServerBuilder} to fix ABI backward compatibility.
+ *
+ * @param <T> The concrete type of this builder.
+ * @see <a href="https://github.com/grpc/grpc-java/issues/7211">grpc/grpc-java#7211</a>
+ */
+public abstract class AbstractServerImplBuilder
```

Comment:
This would need to re-define the methods of ForwardingServerBuilder. At that point extending ForwardingServerBuilder provides nothing. It is good to change ForwardingServerBuilder to `<T extends ServerBuilder>`, though. Although I also wonder if we should remove the class from v1.33.x since it is now unused. (We can just make it package-private to reduce code churn.)

Reply:
Right, that's where I had doubts. On another hand, one could interpret [the spec](https://docs.oracle.com/javase/specs/jls/se7/html/jls-13.html#jls-13.4.4) that it's sufficient for the class to be simply 
present in the hierarchy. I was about to give updated `AbstractManagedChannelImplBuilder` a shot, I already have a setup with the issue reproduced. Will give it a shot.

Reply:
OK, after digging with javap, I understand the order of things with generics now. You're right. I'd have to redefine all methods in `AbstractManagedChannelImplBuilder`, otherwise they are generated with `ServerBuilder` return type. Which is a supertype for `AbstractManagedChannelImplBuilder`, so no match.

Reply:
Good news is the fix for `AbstractManagedChannelImplBuilder` works well.

A client created with gRPC `v1.32.2` is compatible with gRPC jars created from this branch. I confirmed the client is compatible with methods that got moved from abstract to fwd  in `v1.32.2`, then brought back in this PR.  And, just in case, I confirmed that methods that were pushed down from abstract to netty, work too.

```
❯ java -cp "./abi-1.32.2.jar:./deps-1.32.2/*" org.example.AbiClient
ManagedChannelOrphanWrapper{delegate=ManagedChannelImpl{logId=1, target=localhost:100}}
❯ java -cp "./abi-1.32.2.jar:./deps-1.33.0/*" org.example.AbiClient
Exception in thread "main" java.lang.NoSuchMethodError: 
  io.grpc.netty.NettyChannelBuilder.maxRetryAttempts(I)Lio/grpc/internal/AbstractManagedChannelImplBuilder;
	at org.example.AbiClient.<init>(AbiClient.java:14)
	at org.example.AbiClient.main(AbiClient.java:22)
❯ java -cp "./abi-1.32.2.jar:./deps-1.34-be7a6281c/*" org.example.AbiClient
ManagedChannelOrphanWrapper{delegate=ManagedChannelImpl{logId=1, target=localhost:100}}
```
src:
```java
package org.example;

import io.grpc.ManagedChannel;
import io.grpc.ManagedChannelBuilder;
import io.grpc.netty.NettyChannelBuilder;

public class AbiClient {
  private final NettyChannelBuilder ncb;

  public AbiClient() {
    ncb = NettyChannelBuilder
        .forTarget("localhost:100")
        .maxInboundMetadataSize(100)    // own
        .maxRetryAttempts(100);         // Moved from abstract to fwd
  }

  public NettyChannelBuilder getBuilder() {
    return ncb;
  }

  public static void main(String[] args) {
    AbiClient me = new AbiClient();
    ManagedChannel c = me.getBuilder().build();
    System.out.println(c);
  }
}
```